<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Contact Manager</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">
    <item>
     <widget class="QGroupBox" name="groupBox_database">
      <property name="title">
       <string>Database Connection</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_db">
       <item>
        <widget class="QLineEdit" name="lineEdit_host">
         <property name="text">
          <string>localhost</string>
         </property>
         <property name="placeholderText">
          <string>Host (e.g., localhost)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_database">
         <property name="text">
          <string>contacts_db</string>
         </property>
         <property name="placeholderText">
          <string>Database Name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_user">
         <property name="text">
          <string>root</string>
         </property>
         <property name="placeholderText">
          <string>Username</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_password">
         <property name="echoMode">
          <enum>QLineEdit::EchoMode::Password</enum>
         </property>
         <property name="placeholderText">
          <string>Password</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_connect">
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_status">
         <property name="styleSheet">
          <string notr="true">color: red; font-weight: bold;</string>
         </property>
         <property name="text">
          <string>Not Connected</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_actions">
      <property name="title">
       <string>Actions</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_actions">
       <item>
        <widget class="QPushButton" name="pushButton_add">
         <property name="text">
          <string>Add Contact</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_edit">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Edit Contact</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_delete">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Delete Contact</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_fetch">
         <property name="toolTip">
          <string>Fetch a random contact from RandomUser.me API</string>
         </property>
         <property name="text">
          <string>Fetch from API</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_refresh">
         <property name="text">
          <string>Refresh List</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_search">
         <property name="placeholderText">
          <string>Search contacts...</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="tableWidget_contacts">
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <column>
       <property name="text">
        <string>First Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Last Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Email</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Phone</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>City</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Country</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
